src/base/net/downloadhandlerimpl.cpp

```diff
@@ -42,6 +42,16 @@ namespace
 {
     bool saveToFile(const QByteArray &replyData, QString &filePath)
     {
+        if (!filePath.isEmpty())
+        {
+            QFile file {filePath};
+            if (!file.open(QIODevice::WriteOnly))
+                return false;
+
+            file.write(replyData);
+            return true;
```

Comment:
`return (file.write(replyData) == replyData.length());`
IMO both code path should have used that and I'm OK with fixing it in another PR.